DOC Explain how to customise image rendering (#529)
silverstripe/developer-docs: en/08_Changelogs/5.3.0.md

See [convert a file to a different format](/developer_guides/files/file_manipulation/#file-conversion) for more details.

### Improve customisability of rendered images {#image-rendering}

> We've aligned how images added to an HTMLEditorField and SS template images are rendered. The markup up for both `<img>` tag is now generated from the same template file. This will make it easier to customise how images are rendered on your site.

Reviewer: I assume when you say "SS template" you mean "when you invoke the `forTemplate()` method for an image" i.e. `$MyImage`.

Should probably be more explicit about that, especially in the `02_images.md` file. It's very common for people to write explicit `img` tags for each use of an image in a template, so the distinction is important.

Author: I adopted the "invoked in templates" language.

> We've aligned how images added to an HTMLEditorField and SS template images are rendered. The markup up for both `<img>` tag is now generated from the same template file. This will make it easier to customise how images are rendered on your site.
> 
> Simply add a `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` file to your Silverstripe CMS theme.

Reviewer 2: ```suggestion
We've aligned how images added to an [`HTMLEditorField`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorField) and images inside SS templates are rendered. The markup up for both `<img>` tags are now generated from the same template file. This makes it easier to customise how images by creating a `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` file to your Silverstripe CMS theme.
```

Author: I did the first paragraph fix. Not the second.

> Simply add a `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` file to your Silverstripe CMS theme.

Reviewer: Sounds like there's no built-in template. Try something like this:
```suggestion
The template for this is `templates/SilverStripe/Assets/Storage/DBFile_Image.ss`. Add a file with that path to your Silverstripe CMS theme if you want to override it.
```

Author: Done

This opens many interesting possibilities. For example, this code snippet will provide a WebP version of any image rendered on your site.

```html
<picture>

>     <source srcset="{$Convert('webp').Link}" type="image/webp">

Reviewer 2: ```suggestion
    <source srcset="<% $Convert('webp').Link %>" type="image/webp">
```
Can you use those brackets in templates? I would have thought you'd need the regular template "brackets"

Reviewer: ```suggestion
    <source srcset="$Convert('webp').Link" type="image/webp">
```
This is more standard.
The suggestion to use `<%` would give you an error.

Author: I took Guy's suggestion.

    <img $AttributesHTML >
</picture>
```

> You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.

Reviewer: ```suggestion
You can also choose to have different rendering logic for `HTMLEditorField` images and for SS templates images by overriding different templates.
```

Author: Done

> You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.

Reviewer: ```suggestion
You can also choose to have different rendering logic for HTMLEditorField images and for images invoked in templates by overriding different templates.
```
I don't think anywhere else in the docs we say "SS templates"

Author: Done.

> You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.
> 
> - Add a `SilverStripe/Assets/Shortcodes/ImageShortcodeProvider_Image.ss` to your theme to control images added to an HTMLEditorField.
> - Add a `DBFile_Image.ss` file to the root of your theme to control only SS templates images.

Reviewer 2: ```suggestion
You can also choose to have different rendering logic for HTMLEditorField images and for SS template images by overriding different templates.

- Add a `SilverStripe/Assets/Shortcodes/ImageShortcodeProvider_Image.ss` to your theme to control images added to an HTMLEditorField.
- Add a `DBFile_Image.ss` file to the root of your theme to control only SS template images.
```

Author: I used Guy's "invoked in templates" language instead.
